Schema Docs: usdMedia: Create schema docs for SpatialAudio

beersandrew opened the pull request:
### Description of Change(s)

- Create schema docs for SpatialAudio

### Link to proposal ([if applicable](https://openusd.org/release/contributing_to_usd.html#step-1-get-consensus-for-major-changes))

### Fixes Issue(s)

### Checklist

[ ] I have created this PR based on the dev branch

[ ] I have followed the [coding conventions](https://openusd.org/release/api/_page__coding__guidelines.html)

[ ] I have added unit tests that exercise this functionality (Reference: 
[testing guidelines](https://openusd.org/release/api/_page__testing__guidelines.html))

[ ] I have verified that all unit tests pass with the proposed changes

[x ] I have submitted a signed Contributor License Agreement (Reference: 
[Contributor License Agreement instructions](https://openusd.org/release/contributing_to_usd.html#contributor-license-agreement))

beersandrew committed:
feat: create schema doc for SpatialAudio

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,127 @@ over "GLOBAL" (
+and the playback settings (combinations of `startTime`, `endTime`, `playbackMode`, and `mediaOffset`).
+
+The example below shows a typical use case, featuring `Speech` and `Ambient` prims. We can infer from the example the following information:
+- `Speech` is spatially plays from the location of the Cube in the scene, whereas `Ambient` is non-spatial and sounds the same regardles of the listener's position.
```
Maybe "is spatially plays" -> "spatially plays"?

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,127 @@ over "GLOBAL" (
+
+The example below shows a typical use case, featuring `Speech` and `Ambient` prims. We can infer from the example the following information:
+- `Speech` is spatially plays from the location of the Cube in the scene, whereas `Ambient` is non-spatial and sounds the same regardles of the listener's position.
+- `Speech` plays for 10 seconds, based on the math of `endTime` (480) - `startTime` (240) / `timeCodesPerSecond` (24) = 10.
```
Optional, but you could probably omit "the math" and say "...based on: `endTime` (480) -  ....etc"

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,127 @@ over "GLOBAL" (
+            string userDoc = """How the audio is played, spatially or not.
+
+Valid values are: 
+ - `spatial`: Audio is playing from the source of the prim or parented prim. For example this could be used for a character so the audio from the character always comes from their location.
```
Maybe "parent prim" instead of "parented prim"?

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,127 @@ over "GLOBAL" (
         customData = {
-            string userDocBrief = """Brief user doc."""
-            string userDoc = """Longer-form user doc."""
+            string userDocBrief = """Determines when the audio playback should start and stop, also if it should loop."""
```
Minor nit: I would use "and also" rather than "also" here (even though that seems like a lot of "and's"). Or, you could split this -- "Determines when the audio playback should start and stop. Also determines if the audio playback should loop."

dsyu-pixar replied:
Or, thinking about it more, you could split it into general and specific details. Something like: "Determines general rules on the playback of the audio asset. Use this to specify when the audio should start, stop, and if the playback should loop."

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,127 @@ over "GLOBAL" (
+            string userDocBrief = """Determines when the audio playback should start and stop, also if it should loop."""
+            string userDoc = """Determines when the audio playback should start and stop, also if it should loop.
+
+See table below for available values and what they mean relating to if the audio loops and when the audio starts / stops.
```
This table is great! Really helps clarify things.

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,127 @@ over "GLOBAL" (
+| Value                 | Audio Loops?  | StartTime                     | EndTime                                       |
+| --------              | -------       | -------                       | -------                                       |
+| `onceFromStart`       | No            | `startTime` of the Prim       | End of audio file                             |
+| `onceFromStartToEnd`  | No            | `startTime` of the Prim       | `endTime` of the Prim, or End of audio file   |
```
Here, does "End" need to be capitalized?

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,127 @@ over "GLOBAL" (
+            string userDocBrief = """Offset from the start of the audio file for when to start playing, in timeCodes."""
+            string userDoc = """Offset from the start of the audio file for when to start playing, in timeCodes.
+
+Given a 10 second audio file, with a stage `timeCodesPerSecond` value of 24. A `startTime` value of 24.0 will start playing the audio 1 second into the file.            
```
Minor nit: I'd use "....value of 24, a `startTime` of...."  because of the way we start with "Given".

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,127 @@ over "GLOBAL" (
+            string userDocBrief = """Offset from the start of the audio file for when to end playing, in timeCodes."""
+            string userDoc = """Offset from the start of the audio file for when to end playing, in timeCodes.
+
+Given a 10 second audio file, with a stage `timeCodesPerSecond` value of 24. An `endTime` value of 48.0 will end playing the audio 2 seconds into the file.           
```
Same minor nit as startTime

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,127 @@ over "GLOBAL" (
+The example below shows a typical use case, featuring `Speech` and `Ambient` prims. We can infer from the example the following information:
+- `Speech` is spatially plays from the location of the Cube in the scene, whereas `Ambient` is non-spatial and sounds the same regardles of the listener's position.
+- `Speech` plays for 10 seconds, based on the math of `endTime` (480) - `startTime` (240) / `timeCodesPerSecond` (24) = 10.
+- The audio from the `mySpeech.mp3` plays from the 10th second (`startTime` (240) / `timeCodesPerSecond` (24)) to the 20th second (`endTime` (480) / `timeCodesPerSecond` (24)).
```
Optional, but if I'm understanding startTime/endTime correctly, it might be helpful to clarify that this plays from the 10th second of the audio file (as opposed to the 10th second of the scene)

dsyu-pixar reviewed (commented):
Looks great Andy! I added a couple minor comments.

beersandrew committed:
fix: address PR changes

- various grammar errors
- various capitalization errors
- make columns less than 80 characters where possible

beersandrew commented:
thanks for the notes! I think I addressed them all. I also noticed my columns were going past 80 here when they didn't need to, also addressed that.

dsyu-pixar commented on docs/user_guides/schemas/usdMedia/overview.md:
```diff
@@ -0,0 +1,2 @@
+# Overview
```
I think this version of the overview might have gotten accidentally included in the changelist? I'll re-generate the .md's when I merge things into the Pixar codebase, so you don't have to include them if you don't want to.

beersandrew replied:
Sorry, good catch! these slipped through on my side, I removed them now

dsyu-pixar reviewed (approved):
Looks good, thanks for the changes! @beersandrew one question: I'm not certain if the generated markdown files were intended to be added to the PR, but if so, I think the overview.md is out-of-date?

beersandrew committed:
fix: remove generated files

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,154 @@ over "GLOBAL" (
+The example below shows a typical use case, featuring `Speech` and `Ambient` 
+prims. We can infer from the example the following information:
+- `Speech` spatially plays from the location of the Cube in the scene, whereas 
+`Ambient` is non-spatial and sounds the same regardles of the listener's 
```
```suggestion
`Ambient` is non-spatial and sounds the same regardless of the listener's 
```

Missing an "s" -- sorry for not catching this typo earlier!

dsyu-pixar commented on pxr/usd/usdMedia/userDoc/schemaUserDoc.usda:
```diff
@@ -22,50 +22,154 @@ over "GLOBAL" (
+of the audio asset. Use this to specify when the audio should start, stop, 
+and if the playback should loop.
+
+See table below for available values and what they mean relating to if the 
```
Suggestion (that you should feel free to ignore): Maybe simplify the table explanation via something like:

"See the table below for how available values relate to whether the audio loops and when the audio starts / stops."

dsyu-pixar reviewed (approved):
My apologies Andy, I noticed two small things for your review.

beersandrew committed:
fix: address PR comments

- fix typo
- rephrase table description

beersandrew commented:
Appreciate the notes! Updated

dsyu-pixar reviewed (approved):
Thanks Andy! All looks good, will go ahead and merge (and start merging into the Pixar code base)
